agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -149,6 +169,184 @@ impl DeriveSchema for Stage {
             }))
         }
 
+        fn graph_lookup_derive_schema(
+            graph_lookup: &GraphLookup,
+            state: &mut ResultSetState,
+        ) -> Result<Schema> {
+            // it is a bit annoying that we need to clone these Strings just to do a lookup, but I
+            // don't think there is a way around that.
+            let from_field = Namespace(state.current_db.clone(), graph_lookup.from.clone());
+            let mut from_schema = state
+                .catalog
+                .get(&from_field)
+                .ok_or_else(|| Error::UnknownReference(from_field.to_string()))?
+                .clone();
+            if let Some(ref depth_field) = graph_lookup.depth_field {
+                let depth_field_schema = Schema::Atomic(Atomic::Long);
+                insert_required_key_into_document(
+                    &mut from_schema,
+                    depth_field_schema,
+                    vec![depth_field.clone()],
+                );
+            }
+            insert_required_key_into_document(
+                &mut state.result_set_schema,
+                Schema::Array(Box::new(from_schema.clone())),
+                vec![graph_lookup.r#as.to_string()],
```

Comment:
should we just make this _as like in other stages?

Reply:
Yea, I'm not a fan of # fields.